cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -0,0 +1,455 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/bit.cuh>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <rmm/device_buffer.hpp>
+#include <tests/utilities/cudf_gtest.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <iterator>
+#include <memory>
+
+namespace cudf {
+namespace test {
+/**---------------------------------------------------------------------------*
+ * @brief Convenience wrapper for creating a `thrust::transform_iterator` over a
```

Comment:
I think you should document that `UnaryFunction` should be a `__host__` function -- the fact that this uses Thrust might confuse some users.